File: LC/3637_Trionic_Array_I/solution.cpp
class Solution {
public:
    bool isTrionic(vector<int>& nums) {
        int change = 0;
        int n = nums.size();

        if (n < 4 || nums[0] > nums[1]) {
            return false;
        }

        for (int i = 0; i < nums.size()-1; i++) {
            if (change % 2 == 0 && nums[i] > nums[i+1]) {
                change++;
            } else if (change % 2 == 1 && nums[i] < nums[i+1]) {
                change++;
            } else if (nums[i] == nums[i+1]) {
                return false;
            }
        }

        return change == 2;
    }
};

File: LC/3637_Trionic_Array_I/solution.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        change = 0
        n = len(nums)

        if n < 4 or nums[0] >= nums[1]:
            return False
        
        for i in range(n-1):
            if change % 2 == 1 and nums[i] < nums[i+1]:
                change += 1
            elif change % 2 == 0 and nums[i] > nums[i+1]:
                change += 1
            elif nums[i] == nums[i+1]:
                return False

        return change == 2